feat(linter)!: `tsgolint`: request fixes when necessary

camchenry opened the pull request:
With https://github.com/oxc-project/tsgolint/pull/317, tsgolint fixes will become opt-in. So, we need to pass the `-fix` and `-fix-suggestions` when fixes are enabled in oxlint.

@camc314 This will need to be added in oxlint with, or before switching to the next version of tsgolint.

camchenry commented:
@Sysix Is there something obvious I'm missing about why this VSCode test is failing? I can look into it more too, just didn't have time to debug last night. I tried to ensure that the fix kind was set correctly, but maybe I didn't do it right.

camc314 commented on crates/oxc_linter/src/tsgolint.rs:
```diff
@@ -90,6 +113,14 @@ impl TsGoLintState {
                 .stderr(stderr());
 
+            if self.fix {
+                cmd.arg("-fix");
```
fixed

camc314 replied:
thanks for catching 🙂

camchenry committed:
feat(linter)!: `tsgolint`: request fixes when necessary (#15048)

With https://github.com/oxc-project/tsgolint/pull/317, tsgolint fixes will become opt-in. So, we need to pass the `-fix` and `-fix-suggestions` when fixes are enabled in oxlint.

@camc314 This will need to be added in oxlint with, or before switching to the next version of tsgolint.
